src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -111,8 +111,8 @@ internal partial class DiagnosticIncrementalAnalyzer
                         // Analyzers are disabled for all documents.
                         BackgroundAnalysisScope.None => false,
 
-                        // Analyzers are enabled for active document.
-                        BackgroundAnalysisScope.ActiveFile => isActiveDocument,
+                        // Analyzers are enabled for visible documents and open documents which had errors/warnings in prior snapshot.
+                        BackgroundAnalysisScope.VisibleFilesAndFilesWithPreviouslyReportedDiagnostics => IsVisibleDocumentOrOpenDocumentWithPriorReportedVisibleDiagnostics(isVisibleDocument, isOpenDocument, previousData),
```

Comment:
This is the core fix - we now have consistent semantics between compiler and analyzer diagnostics refresh for `Current Document` scope.

Rest of the changes in this PR are just cascading rename changes as the enum field name has also been renamed to match the field name within `CompilerDiagnosticScope` enum